site/content/3.12/aql/high-level-operations/upsert.md

```diff
@@ -240,18 +247,15 @@ transactional, especially if it involves different shards and/or DB-Servers.
   `exclusive` option to limit write concurrency for this collection to 1, which
   helps avoiding conflicts but is bad for throughput!
 
-- Using very large transactions in an UPSERT (e.g. UPSERT over all documents in
-  a collection) an **intermediate commit** can be triggered. This intermediate
-  commit will write the data that has been modified so far. However this will
-  have the side-effect that atomicity of this operation cannot be guaranteed
-  anymore and that ArangoDB cannot guarantee to that read your own writes in
-  upsert will work.
+- `UPSERT` operations do not observe own writes correctly in cluster deployments
+  for collections with more than a single shard, and also not if you use OneShard
+  but disable the `cluster-one-shard` optimizer rule.
```

Comment:
I guess that nobody will explicitly disable the `cluster-one-shard` optimizer rule. But the key thing here is that this rule does not exist in the Community Edition, so it is disabled there in every case.